### ISSUE TITLE
Relationship field with posts of more than one post type

### ISSUE THREAD

**Issue by pmandiola** [pods-framework/pods#5150]
## Issue Overview
Right now when setting up a relationship field from a Pod connecting to another Pod or a regular post, I have to choose a specific CPT type. In some situations, it would be very useful to be able to set up the relationship field to be with posts of more than one post type, or even set that posts of any post type could be selected for the relationship.
My current workaround for this is saving the related post id as a regular post meta, but it can't be done through a Pods relationship.

## Expected Behavior
Allow setting up a Pods relationship field pointing to more than one post type, so then I can assign posts of different post types to that relationship.

## Current Behavior
Only one specific post type can be selected.

**Comment by quasel** [comment]
What's your use case? 
I'm not sure that's easily doable due to the way relationships currently work but for sure an idea!

**Comment by pmandiola** [comment]
My use case is a bit complicated but I'll try to explain:

I'm developing a website that has some functionalities similar to a regular social network. Users can upload/record their own videos directly from the website (a CPT called "recording"), and they can also browse a video library maintained by us (a CPT called "video"). There is also a 3rd CPT more similar to a regular wordpress post with text and some images, called "recommendation". And we'll be adding more in the future for different content.

Users can share their own recordings, videos, or recomendations with other users in 1-1 conversations or groups. When a user shares something, I'm using another CPT called "share", which has a sharing message, with whom its is shared, and the shared post. This shared post is the relationship field I need to set to any kind of post type, as I then display the "share" differently depending on the CPT that was shared. Think of Facebook posts sharing other Facebook posts.

I hope it made some sense!

**Comment by quasel** [comment]
makes sense - but I think this might be a bit out of scope :/ what about using Gutenberg ;) and store it all in the same CPT :D 

@sc0ttkclark something to look at with 2.8?

**Comment by sc0ttkclark** [comment]
I'd love to make this possible for post type or taxonomy based relationships.

It's tricky but I think a lot of the stuff done in Pods 2.8 should enable us make this possible.
